tests/acceptance/add-new-item-to-inventory-test.js

```diff
@@ -0,0 +1,91 @@
+
+var App, location1, designation, code;
+
+module('Acceptance: Add item to inventory', {
+  beforeEach: function() {
+    App = startApp({}, 2);
+    TestHelper.setup();
+    visit("/");
+    location1 = FactoryGuy.make("location");
+    designation = FactoryGuy.make("designation", { state: "closed" });
+    mockFindAll('designation').returns({json: {designations: [designation.toJSON({includeId: true})]}});
+    mockFindAll('location').returns({json: {locations: [location1.toJSON({includeId: true})]}});
+    code = FactoryGuy.make("code", {location: location1});
+  },
+  afterEach: function() {
+    Ember.run(function() { TestHelper.teardown(); });
+    Ember.run(App, 'destroy');
+  }
+});
+
+test("Redirect to /search_code after clicking Add item to inventory", function(assert) {
+  assert.expect(14);
+
+  $.mockjax({url: '/api/v1/package_type*', type: 'GET', status: 200,responseText: {
+      codes: [code.toJSON({includeId: true})]
+    }
+  });
+  click($('.center-text a'));
+  andThen(function() {
+    assert.equal(currentPath(), "search_code");
+    fillIn("#searchText", code.get('name'));
+    click($('.list li:first')[0]);
+    $.mockjax({url:"/api/v1/inventory*", type: 'POST', status: 200,responseText:{"inventory_number":"000311"}});
+    $.mockjax({url:"/api/v1/images/generate_sign*", type: 'GET', status: 200,responseText:{"api_key":123456789876543,"signature":"3ec17bf700bc23446d61932385d","timestamp":1234567891,"tags":"staging"}});
+    andThen(function() {
+      assert.equal(currentPath(), "items.new");
+      //fillIn("#searchText", "Baby Crib");
+      assert.equal($('#qty').val(), '1');
+      assert.equal($('.columns .left input').last().val(), code.get('name'));
+      assert.equal($('#description').val(), code.get('name'));
+      assert.equal($('#height').val(), "");
+      assert.equal($('#width').val(), "");
+      assert.equal($('#length').val(), "");
+      assert.equal($('label select').val(), "B");
+      assert.equal($('#condition-select').val(), "U");
+      assert.equal($('.small-9.columns input').first().val(), "");
+      assert.equal($('.small-9.columns input').last().val(), location1.get('building')+location1.get('area'));
+      assert.equal($('.inventory-number').text().match(/\d+/g)[0], "000311");
+    });
+    andThen(function() {
+        click($('.button.expand').last());
+
+        $.mockjax({url:"/api/v1/package*", type: 'POST', status: 200,responseText:{
+          "item" :  {"id":971,"quantity":1,"length":null,"width":null,"height":null,"notes":"Baby Crib, Set (frame, mattress)","inventory_number":"000317","created_at":"2017-05-19T11:50:42.179786","updated_at":"2017-05-19T11:50:42.179786","item_id":null,"is_set":false,"grade":"B","designation_name":null,"sent_on":null,"code_id":9,"image_id":null,"donor_condition_id":3,"set_item_id":null,"has_box_pallet":false,"case_number":"","allow_web_publish":null,"received_quantity":1,"package_type_id":9,"packages_location_ids":[764],"orders_package_ids":[]},
```

Comment:
For this response use item, code, location, packages_location etc made by Factory.